src/widgets/DataTable/TextInputCell.js

```diff
@@ -0,0 +1,140 @@
+ * @param {string|number} rowKey     Unique key associated to row cell is in
+ * @param {string|number} columnKey  Unique key associated to column cell is in
+ * @param {bool} isDisabled          If `true` will render a plain Cell element with no interaction
+ * @param {String} placeholderColour Placholder text colour
+ * @param {func}  dispatch           Reducer dispatch callback for handling actions
+ * @param {Object}  viewStyle        Style object for the wrapping View component
+ * @param {Object}  textStyle        Style object for the inner Text component
+ * @param {Object}  textInputStyle   Style object for TextInput component.
+ * @param {Number}  width            Optional flex property to inject into styles.
+ * @param {bool}    debug            Logs rendering of this component.
+ * @param {string}  keyboardType     Type of keyboard for the TextInput.
+ * @param {string}  placeholder      placeholder text
+ * @param {Object} cellTextStyle     text style for the disabled Cell component.
+ * @param {Bool}  isLastCell         Indicator if this cell is last in a row,
+ *                                   removing the borderRight,
+ * @param {func}  editAction         Action creator for handling editing of this cell.
+ *                                   `(newValue, rowKey, columnKey) => {...}`
+ */
+const TextInputCell = React.memo(
+  ({
+    value,
+    rowKey,
+    columnKey,
+    isDisabled,
+    placeholderColour,
+    editAction,
+    dispatch,
+    isLastCell,
+    width,
+    debug,
+    keyboardType,
+    placeholder,
+    viewStyle,
+    rowIndex,
+    textInputStyle,
+    cellTextStyle,
+  }) => {
+    if (debug) console.log(`- TextInputCell: ${value}`);
+
+    const usingPlaceholder = placeholder && !value;
+
+    const { focusNextCell, getRefIndex, getCellRef } = React.useContext(RefContext);
+    const refIndex = getRefIndex(rowIndex, columnKey);
+
+    const onEdit = newValue => dispatch(editAction(newValue, rowKey, columnKey));
+    const focusNext = () => focusNextCell(refIndex);
+
+    // Render a plain Cell if disabled.
+    if (isDisabled) {
+      return (
+        <Cell
+          key={columnKey}
+          viewStyle={viewStyle}
+          textStyle={cellTextStyle}
+          value={value}
+          width={width}
+          isLastCell={isLastCell}
+        />
+      );
+    }
+
+    const internalViewStyle = getAdjustedStyle(viewStyle, width, isLastCell);
+    const internalTextStyle = getAdjustedStyle(textInputStyle, width);
+
+    // Render a Cell with a textInput.
+    return (
+      <View style={internalViewStyle}>
+        <TextInput
+          ref={getCellRef(refIndex)}
+          placeholder={placeholder}
+          style={internalTextStyle}
+          value={usingPlaceholder ? '' : String(value)}
+          placeholderTextColor={placeholderColour}
+          onChangeText={onEdit}
+          onSubmitEditing={focusNext}
+          underlineColorAndroid="#CDCDCD"
```

Comment:
Shouldn't the colours come from `colors.js`? ( this and in line 137 )
or does that not matter for some things?

Reply:
all of this folder is being shifted to `react-native-data-table` repo when it's done , which it won't have access to. I've added the colour as a prop instead, with that colour as a default de60b21bd620ee9e1d5ce9d35fc617fffec50bd9. Maybe when we shift this to the repo we can see if a `styles` file is needed here #1137